internal/backend/runtime/omni/migration/migration_test.go

```diff
@@ -1798,6 +1798,138 @@ func (suite *MigrationSuite) TestMoveEtcdBackupStatuses() {
+	ctx, cancel := context.WithTimeout(suite.T().Context(), 10*time.Second)
+	defer cancel()
+
+	clusterConfigVersion12 := omni.NewClusterConfigVersion(resources.DefaultNamespace, "cluster-12")
+	clusterConfigVersion12.TypedSpec().Value.Version = "v1.2.0"
+
+	clusterConfigVersion14 := omni.NewClusterConfigVersion(resources.DefaultNamespace, "cluster-14")
+	clusterConfigVersion14.TypedSpec().Value.Version = "v1.4.8"
+
+	// patch expected: apidCheckExtKeyUsage needs to be preserved
+	cluster12config1 := omni.NewClusterMachineConfig(resources.DefaultNamespace, "cluster-12-machine-1")
+	cluster12config1.Metadata().Labels().Set(omni.LabelCluster, "cluster-12")
+
+	suite.Require().NoError(cluster12config1.TypedSpec().Value.SetUncompressedData([]byte(`machine:
+  features:
+    apidCheckExtKeyUsage: true
+    foo: bar
+`)))
+
+	// no patch expected: it does not have any of the features to be preserved
+	cluster12config2 := omni.NewClusterMachineConfig(resources.DefaultNamespace, "cluster-12-machine-2")
+	cluster12config2.Metadata().Labels().Set(omni.LabelCluster, "cluster-12")
+
+	suite.Require().NoError(cluster12config2.TypedSpec().Value.SetUncompressedData([]byte(`machine:
+  features:
+    foo: bar"
+`)))
+
+	// patch expected: **both** apidCheckExtKeyUsage and diskQuotaSupport need to be preserved
+	cluster12config3 := omni.NewClusterMachineConfig(resources.DefaultNamespace, "cluster-12-machine-3")
+	cluster12config3.Metadata().Labels().Set(omni.LabelCluster, "cluster-12")
+
+	suite.Require().NoError(cluster12config3.TypedSpec().Value.SetUncompressedData([]byte(`machine:
+  features:
+    foo: bar
+    apidCheckExtKeyUsage: true
+    baz: qux
+    diskQuotaSupport: true
+    quux: corge
+`)))
+
+	// patch expected: **only** diskQuotaSupport needs to be preserved
+	cluster14config1 := omni.NewClusterMachineConfig(resources.DefaultNamespace, "cluster-14-machine-1")
+	cluster14config1.Metadata().Labels().Set(omni.LabelCluster, "cluster-14")
+
+	suite.Require().NoError(cluster14config1.TypedSpec().Value.SetUncompressedData([]byte(`machine:
+  features:
+	foo: bar
+    diskQuotaSupport: true
+    apidCheckExtKeyUsage: true
+`)))
+
+	// no patch expected: it does not have any of the features to be preserved
+	cluster14config2 := omni.NewClusterMachineConfig(resources.DefaultNamespace, "cluster-14-machine-2")
+	cluster14config2.Metadata().Labels().Set(omni.LabelCluster, "cluster-14")
+
+	suite.Require().NoError(cluster14config2.TypedSpec().Value.SetUncompressedData([]byte(`machine:
+  features:
+	foo: bar
+`)))
+
+	// no patch expected, since it will be in tearing down phase
+	cluster14config3 := omni.NewClusterMachineConfig(resources.DefaultNamespace, "cluster-14-machine-3")
+	cluster14config3.Metadata().Labels().Set(omni.LabelCluster, "cluster-14")
+
+	suite.Require().NoError(cluster14config1.TypedSpec().Value.SetUncompressedData([]byte(`machine:
+  features:
+	foo: bar
+    diskQuotaSupport: true
```

Comment:
I wonder if that's how displays it, or the idents are messed up here?

Reply:
Yep, thanks, copy/pasting messed it up in a few places. Now enabled displaying whitespace in my editor so it won't happen in the future.